actors/miner/src/lib.rs

```diff
@@ -878,6 +879,481 @@ impl Actor {
+
+            if update.replica_proof.len() > 4096 {
+                info!(
+                    "update proof is too large ({}), skipping sector {}",
+                    update.replica_proof.len(),
+                    update.sector_number,
+                );
+                continue;
+            }
+
+            if update.deals.is_empty() {
+                info!(
+                    "must have deals to update, skipping sector {}",
+                    update.sector_number,
+                );
+                continue;
+            }
+
+            if update.deals.len() as u64 > sector_deals_max(info.sector_size) {
+                info!(
+                    "more deals than policy allows, skipping sector {}",
+                    update.sector_number,
+                );
+                continue;
+            }
+
+            if update.deadline >= WPOST_PERIOD_DEADLINES {
+                info!(
+                    "deadline {} not in range 0..{}, skipping sector {}",
+                    update.deadline, WPOST_PERIOD_DEADLINES, update.sector_number
+                );
+                continue;
+            }
+
+            // Skip checking if CID is defined because it cannot be so in Rust
+
+            if !is_sealed_sector(&update.new_sealed_cid) {
+                info!(
+                    "new sealed CID had wrong prefix {}, skipping sector {}",
+                    update.new_sealed_cid, update.sector_number
+                );
+                continue;
+            }
+
+            // If the deadline is the current or next deadline to prove, don't allow updating sectors.
+            // We assume that deadlines are immutable when being proven.
+            if !deadline_is_mutable(
+                state.current_proving_period_start(rt.curr_epoch()),
+                update.deadline,
+                rt.curr_epoch(),
+            ) {
+                info!(
+                    "cannot upgrade sectors in immutable deadline {}, skipping sector {}",
+                    update.deadline, update.sector_number
+                );
+                continue;
+            }
+
+            if let Ok(healthy) = state.check_sector_active(
+                rt.store(),
+                update.deadline,
+                update.partition,
+                update.sector_number,
+                true,
+            ) {
+                if !healthy {
+                    info!(
+                        "sector isn't healthy, skipping sector {}",
+                        update.sector_number
+                    );
+                    continue;
+                }
+            } else {
+                return Err(actor_error!(
+                    ErrIllegalArgument,
+                    "error checking sector health"
+                ));
+            }
+
+            // TODO: must_get has never been used before, so check impl
+            let res = Sectors::must_get(&sectors, update.sector_number);
```

Comment:
A bit late now, but `must_get` is for cases where not having the sector in the state is a state error. The way it is now, we'll "skip" forward even if the state is somehow corrupted (which shouldn't happen, but it would be nicer to fail hard in that case).